### PR TITLE
Add date range CSV exporting for RSVPs

### LINKED ISSUE #28: [HIGH PRIORITY] Tracking system/report for Event Attendance
- Currently there is only a way to track rsvps, not actual attendance, so we don’t have accurate counts when we export everyone’s data. 

- We would like a way to make an Attendance Report with parameters for a specific date period but also for a specific user.

### PR DESCRIPTION
**Pull request by JasonTheAdams**
Resolves #28 

## Description

This improves the Tools > Export utility for RSVPs. Instead of exporting the large XML file, it provides a _much_ faster and simpler CSV file that can be used in any spreadsheet program. The name and date of the event RSVP'd for are also included for each line.

It also gives start and end date to limit the range of RSVPs to download. By default the export range is the last 30 days.

## Visuals

image

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@mathetos This is ready for review. I privately sent you an example export. Please let me know if this provides all the needed data.

File: rsvps/_rsvp.php [hunk 1]
```diff
@@ -33,18 +34,19 @@ function get_rsvps(array $args = []): array
...
+ * @param int|int[] $event_id
+ * @param array     $args additional WP_Query args to add or overload
  *
  * @return WP_Post[]|int[]
  */
-function get_event_rsvps(int $event_id, array $args = []): array
+function get_event_rsvps($event_id, array $args = []): array
 {
-    $meta_query = $args['meta_query'] ?: [];
+    $meta_query = $args['meta_query'] ?? [];
```
Comment (mathetos) [inline rsvps/_rsvp.php:44]: TIL about the "Null Coalescing operator" ... I would have thought this was a typo, but knowing you, I researched first. Interesting!
https://www.tutorialspoint.com/what-does-double-question-mark-operator-mean-in-php

File: rsvps/exporter/Exporter.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    {
+        add_action('export_filters', [$this, 'renderExportFilters']);
+        add_action('export_wp', [$this, 'generateCSV']);
+    }
+
+    public function renderExportFilters(): void
+    {
+        wp_enqueue_script('jquery-ui-datepicker');
+        wp_enqueue_style('jquery-ui-datepicker-style',
+            '//ajax.googleapis.com/ajax/libs/jqueryui/1.10.4/themes/smoothness/jquery-ui.css');
```
Comment (mathetos) [inline rsvps/exporter/Exporter.php:15]: Do we actually need to enqueue this stylesheet? Doesn't WP have the datepicker styles built-in somewhere?

Comment (JasonTheAdams) [reply rsvps/exporter/Exporter.php:15]: I was surprised, too, but it's necessary. Without this the datepicker was missing styles and had no background. They're only enqueued on this one page so it shouldn't have any negative effects elsewhere.

**Comment by mathetos** [comment]
@JasonTheAdams 

This looks great and is super fast. Love it!

**Pull request merged**